pkg/mountutil/volumestore/volumestore.go

```diff
@@ -83,13 +84,30 @@ func (vs *volumeStore) Create(name string, labels []string) (*native.Volume, err
 	}
 	volPath := filepath.Join(vs.dir, name)
 	volDataPath := filepath.Join(volPath, DataDirName)
-	fn := func() error {
+	volFilePath := filepath.Join(volPath, volumeJSONFileName)
+	fn := func() (err error) {
 		if err := os.Mkdir(volPath, 0700); err != nil {
 			return err
 		}
+		defer func() {
+			if err != nil {
+				os.Remove(volPath)
+			}
+		}()
 		if err := os.Mkdir(volDataPath, 0755); err != nil {
 			return err
 		}
+		defer func() {
+			if err != nil {
+				if _, statErr := os.Stat(volFilePath); statErr != nil && !os.IsNotExist(statErr) {
```

Comment:
IMO it may be more readable with a defer func that remove `volFilePath`  just before `os.WriteFile(volFilePath, labelsJSON, 0644)`

Reply:
> IMO code should be more readable by doing thing step by step: with a seperate defer func that removes `volFilePath` just before `os.WriteFile(volFilePath, labelsJSON, 0644)`

Fixed.
cc @ktock